src/main/java/org/spongepowered/common/mixin/core/world/gen/ChunkProviderServerMixin.java

```diff
@@ -268,22 +270,34 @@ public boolean tick()
             final Iterator<Chunk> iterator = this.loadedChunks.values().iterator();
             int chunksUnloaded = 0;
             final long now = System.currentTimeMillis();
+            final long world_time = this.world.getTotalWorldTime();
             while (chunksUnloaded < this.impl$maxChunkUnloads && iterator.hasNext()) {
                 final Chunk chunk = iterator.next();
                 final ChunkBridge spongeChunk = (ChunkBridge) chunk;
-                if (chunk != null && chunk.unloadQueued && !spongeChunk.bridge$isPersistedChunk()) {
+                if (chunk == null || spongeChunk.bridge$isPersistedChunk()) {
+                    continue;
+                }
+                if (chunk.unloadQueued) {
                     if (this.bridge$getChunkUnloadDelay() > 0) {
                         if ((now - spongeChunk.bridge$getScheduledForUnload()) < this.impl$chunkUnloadDelay) {
                             continue;
                         }
                         spongeChunk.bridge$setScheduledForUnload(-1);
                     }
                     chunk.onUnload();
-                    this.saveChunkData(chunk);
-                    this.saveChunkExtraData(chunk);
+                }
+                // max lifetime only applies to chunks that should stay loaded but still need saving once in a while
+                else if (this.impl$maxChunkLifetime <= 0 // if setting is disabled
```

Comment:
Just clarify something for me here: this seems to change the logic such that chunk data and extra chunk data will never be saved if the max chunk lifetime is disabled because this check will then always succeed and fall through to the continue? Shouldn't this check only pass if the chunk isn't due to be unloaded in the first place?

Reply:
Oh, wait, I see, else if. Got it.

Reply:
Exactly. If the chunk is scheduled for unload - it's lifetime doesn't need to be checked - we just fall through to the saving code.